### PR TITLE
Check Turn / Then Banner on each update

### PR DESCRIPTION
**Pull request by danesfeder**
- Previously would only check when new step was found 
- Fixes bug with then banner not showing after medium alert level

### CODE REVIEW

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionStepResources.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ private void thenStep(LegStep upcomingStep, LegStep followOnStep, double current
     thenStepManeuverType = followOnStep.maneuver().type();
     thenStepManeuverModifier = followOnStep.maneuver().modifier();
     // Should show then step if the upcoming step is less than 25 seconds
```
Comment (Guardiola31337) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionStepResources.java]: This comment is now lying. This is an example of why I'm not a big fan of comments :trollface:

Comment (Guardiola31337) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionStepResources.java]: > The best comment is a good name for a method or class.

or _variable_ in this case 😛
👀 Code Smells 👉 [Comments](https://refactoring.guru/smells/comments)

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionStepResources.java [hunk 2]
```diff
@@ -98,7 +98,7 @@ private void thenStep(LegStep upcomingStep, LegStep followOnStep, double current
...
-    shouldShowThenStep = upcomingStep.duration() <= (25d * 1.2d) && currentDurationRemaining <= (70d * 1.2d);
+    shouldShowThenStep = upcomingStep.duration() <= (25d * 1.2d) && currentDurationRemaining <= 70d;
```
Comment (Guardiola31337) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionStepResources.java]: Could we fix the variable name to reflect what actually is doing? This way comment will become superfluous so it won't be necessary.

Comment (Guardiola31337) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionStepResources.java]: What about [decomposing conditional](https://refactoring.com/catalog/decomposeConditional.html) extracting that complicated expression into a well-named `private` method? Conditionals tend to get more and more complicated in their logic over time.

Comment (Guardiola31337) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionStepResources.java]: Magic numbers.

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionView.java [hunk 3]
```diff
@@ -667,10 +670,12 @@ private boolean newStep(RouteProgress routeProgress) {
...
   private void updateTurnLanes(InstructionModel model) {
-    if (model.getStepResources().getTurnLanes() != null
-      && !TextUtils.isEmpty(model.getStepResources().getManeuverViewModifier())) {
-      turnLaneAdapter.addTurnLanes(model.getStepResources().getTurnLanes(),
-        model.getStepResources().getManeuverViewModifier());
+    List<IntersectionLanes> turnLanes = model.getStepResources().getTurnLanes();
+    String maneuverViewModifier = model.getStepResources().getManeuverViewModifier();
+    double durationRemaining = model.getProgress().currentLegProgress().currentStepProgress().durationRemaining();
+
+    if (turnLanes != null && !TextUtils.isEmpty(maneuverViewModifier) && durationRemaining <= 70d) {
```
Comment (Guardiola31337) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionView.java]: What about simplifying this conditional? 👀 [Decompose Conditional](https://sourcemaking.com/refactoring/decompose-conditional)

Comment (Guardiola31337) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionView.java]: Magic number.

**Comment by danesfeder** [comment]
@Guardiola31337 Comments addressed 🍍

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionStepResources.java [hunk 4]
```diff
@@ -97,8 +100,12 @@ private void intersectionTurnLanes(LegStep step) {
   private void thenStep(LegStep upcomingStep, LegStep followOnStep, double currentDurationRemaining) {
     thenStepManeuverType = followOnStep.maneuver().type();
     thenStepManeuverModifier = followOnStep.maneuver().modifier();
-    // Should show then step if the upcoming step is less than 25 seconds
-    shouldShowThenStep = upcomingStep.duration() <= (25d * 1.2d) && currentDurationRemaining <= (70d * 1.2d);
+    shouldShowThenStep = validStepDuration(upcomingStep, currentDurationRemaining);
+  }
+
+  private boolean validStepDuration(LegStep upcomingStep, double currentDurationRemaining) {
```
Comment (Guardiola31337) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionStepResources.java]: Minor detail, what about `isValidStepDuration` instead?

**Pull request merged**